crates/kitsune_p2p/kitsune_p2p/src/metrics.rs

```diff
@@ -76,9 +170,46 @@ impl<'lt> AgentLike<'lt> {
 }
 
 impl Metrics {
-    #[cfg(test)]
-    pub fn new() -> Self {
-        Self::default()
+    /// Sucessful and unsuccessful messages from the remote
+    /// can be combined to estimate a "reachability quotient"
+    /// between 1 (or 0 if empty) and 100. Errors are weighted
+    /// heavier because we retry less frequently.
+    /// Call this to register a reachability event.
+    /// Note, `record_success` and `record_error` below invoke this
+    /// function internally, you don't need to call it again.
+    pub fn record_reachability_event<'a, T, I>(&mut self, success: bool, remote_agent_list: I)
+    where
+        T: Into<AgentLike<'a>>,
+        I: IntoIterator<Item = T>,
+    {
+        for agent_info in remote_agent_list {
+            let info = self
+                .map
+                .entry(agent_info.into().agent().clone())
+                .or_default();
+            if success {
+                info.reachability_quotient.push(100);
+            } else {
+                info.reachability_quotient.push_n(1, 5);
```

Comment:
why don't we just push `0` 5 times instead of `1`?

Reply:
this is an algorithm hold-over from when it was a histogram... the hdr histogram algorithm can't understand zeroes, so they get clamped to 1 anyways... with the running average here, we could totally do zero